### PR TITLE
Add radio module, async function handling

### PR DESCRIPTION
**Pull request by zuzak**
This PR adds commands to see what's currently playing on BBC Radio stations:
```
02:51:23 <@zuzak> !r1
02:51:24 < civilservant-4> zuzak: Now: Annie Nightingale presents... (Cosmic Boys and A-Trak) followed by Radio 1 Comedy in 9 minutes (3:00 am)
02:51:31 <@zuzak> !r2
02:51:32 < civilservant-4> zuzak: Now: OJ Borg (No Woman, No Cry by Fugees) followed by Sounds Of The 80s with Gary Davies in 8 minutes (3:00 am)
02:51:37 <@zuzak> !r3
02:51:38 < civilservant-4> zuzak: Now: Through the Night (Symphony No 4 in B flat major, Op 60 by Ludwig van Beethoven) followed by Breakfast in 4 hours (6:30 am)
02:51:49 <@zuzak> !r4
02:51:50 < civilservant-4> zuzak: Now: Selection of BBC World Service Programmes (20/03/2019) followed by Shipping Forecast in 2 hours (5:20 am)
```

(it would support any BBC radio station, but I didn't bother adding them all.)

This PR also adds support for returning promises -- you will now be able to write asynchronous command functions and the bot will be clever enough to resolve the promise for you.

### CODE REVIEW

**Comment by zuzak** [comment]
I don't particularly care what codefactor thinks

File: modules/_commands.js [hunk 1]
```diff
@@ -53,18 +66,10 @@ module.exports = {
...
-                console.log('No return for ' + cmd)
-              } else if (typeof output === 'string') {
-                return bot.say(msg.to, msg.nick + ': ' + output)
-              } else if (output.constructor === Array) {
-                for (var k = 0; k < output.length; k++) {
-                  if (typeof output[k] === 'string') {
-                    return bot.say(msg.to, msg.nick + ': ' + output[k])
-                  }
-                }
+              if (cmd.constructor.name === 'AsyncFunction') {
```
Comment (danhedron) [inline modules/_commands.js]: `Promise.resolve(cmd(bot, msg))` can be used here to ensure the result is a Promise that you can `.then(...)` even if the `cmd()` returned not-a-promise.

File: modules/radio.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+const request = require('request-promise')
+const moment = require('moment')
+const colors = require('irc').colors
+
+const radio = async (station) => {
+  const url = 'https://polling.bbc.co.uk/modules/onairpanel/include/' + station + '.json'
```
Comment (danhedron) [inline modules/radio.js]:
```suggestion
  const url = `https://polling.bbc.co.uk/modules/onairpanel/include/${station}.json`
```

File: modules/_commands.js [hunk 3]
```diff
@@ -1,6 +1,19 @@
 /**
  * Main logic for the other commands live here.
  */
+const processOutput = (bot, msg, output) => {
+  if (typeof output === 'string') {
+    return bot.say(msg.to, msg.nick + ': ' + output)
+  } else if (output.constructor === Array) {
+    for (var k = 0; k < output.length; k++) {
```
Comment (danhedron) [inline modules/_commands.js]:
```js
output.forEach(line => {
  if (typeof line === 'string') bot.say(msg.to, `${msg.nick}: ${line}`)
})
```

**Pull request merged**